src/shared/sleep-config.c

```diff
@@ -102,6 +107,179 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         return 0;
 }
 
+/* If battery percentage capacity is less than equal to 5% return success */
+int battery_is_low(void) {
+        int r;
+
+        r = read_battery_capacity_percentage();
+        if (r == -ENOENT)
+               return false;
+        if (r < 0)
+               return r;
+
+        return r <= 5;
+        /* If battery current capacity percentage is equal to or less than 5%, return 1.
+         * We have not used battery capacity_level since value is set to full
+         * or Normal in case acpi is not working properly.
+         * In case of no battery 0 will be returned and system will be suspended for 1st cycle then hibernated */
+}
+
+/* battery percentage capacity fetched from capacity file and if in range 0-100 then returned */
+int read_battery_capacity_percentage(void) {
+        _cleanup_free_ char *bat_cap = NULL;
+        int battery_capacity, r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
+        if (r == -ENOENT)
+               return log_debug_errno(r, "/sys/class/power_supply/BAT0/capacity is unavailable. Assuming no battery exists: %m");
+               /* Handling case when no battery is present. Exception here file is manually deleted / corrupted */
```

Comment:
First comment, then write code:
```c
if (r == -ENOENT)
        /* comment */
        return log_debug_errno(r, "...");
```

BTW, the comment seems to be outdated. Now this function does not have special handling of -ENOENT, but its caller handle it gracefully. Hence, I'd like this simply:
```
r = read_one_line_file(...);
if (r < 0)
        return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
```

Reply:
Yeah. You are right. Fixed it.